In [1]:
from astroquery.vizier import Vizier
from sqlalchemy import exc

## Catalogue selection

Database of variable stars identified by project OGLE published by Zebrun et al. (2001) Acta Astronomica v.51,  pp. 317

Vizier ID of article is *J/AcA/51/317/ogle2mc*

In [3]:
Vizier.ROW_LIMIT = -1
catalog_name = 'J/AcA/51/317/ogle2mc'
catalog = Vizier.get_catalogs(catalog_name)
catalog.pprint

<bound method TableList.pprint of TableList with 1 tables:
	'0:J/AcA/51/317/ogle2mc' with 16 column(s) and 68194 row(s) >

Output is in the form *TableList* - list of astropy Table (see https://docs.astropy.org/en/stable/api/astropy.table.Table.html)

Table contains 68194 rows, one row per one variable star. Columns contains information about stars like RA,DEC coordinates, magnitudes, observational erros and other observed quantities or useful informations. We can print all column names by using

In [4]:
print(catalog[0].colnames)

['OGLE', 'RAJ2000', 'DEJ2000', 'MC', 'Fld', 'Idat', 'LC', 'ID1', 'dist1', 'mag1', 'e_mag1', 'ID2', 'dist2', 'mag2', 'e_mag2', 'Remarks']


- **OGLE** is ID number (constructed from coordinates) originally used by OGLE project
- **RAJ2000,DEJ2000** are coordinates
- **MC** is flag for Large Magellanic Cloud or Small Magellanic Cloud
- **Fld** is observed field number from OGLE project
- **Idat** Number of I-band data from OGLE database
- **ID1** DOPHOT ID number of the closest star identified by DOPHoT on the DIA reference frame
- **dist1** Distance to DOPHOT star **ID1**
- **mag1** Magnitude of DOPHOT star **ID1**
- **e_mag1** Error on **mag1**
- **ID2** OGLE ID number of the closest star identified by OGLE
- **dist2** Distance to OGLE star ientified by OGLE template
- **mag2** Magnitude of OGLE star **ID2**
- **e_mag2** Error on **mag2**

As a test we tried to find information about star *OGLE05201427-7007382* and for our purpose we reduce columns only on **OGLE,RAJ2000,DEJ2000,MC,Fld,Idat,ID2,mag2,e_mag2**.

In [4]:
result=Vizier(columns=['RAJ2000','DEJ2000','Idat','ID2','MC','Fld','mag2']).query_constraints(catalog=catalog_name,OGLE='==05201427-7007382')
result.pprint

<bound method TableList.pprint of TableList with 1 tables:
	'0:J/AcA/51/317/ogle2mc' with 7 column(s) and 1 row(s) >

Result in the form of *TableList* contains Table with the selected information about the star.

In [7]:
table=result[0]
table

<Table masked=True length=1>
 RAJ2000   DEJ2000   Idat  ID2   MC   Fld    mag2 
   deg       deg                             mag  
 float64   float64  int16 int32 str1 int16 float32
--------- --------- ----- ----- ---- ----- -------
 80.05945 -70.12728   273 45899    L    21  17.802

Coordinates of the **RAJ2000,DEJ2000** of the star are in the unit *deg*. It is useful to have coordinates in other format

In [10]:
from astropy import units as u
from astropy.coordinates import SkyCoord

coords_star = SkyCoord(table['RAJ2000'].data*u.deg,table['DEJ2000'].data*u.deg)
coords_star.to_string('hmsdms')



[u'05h20m14.268s -70d07m38.208s']

Similarly we can query catalog of LMC Be stars published by *Sabogal et al. (2005)* in Monthly Notices of the Royal Astronomy

Vizier ID of the catalog is *J/MNRAS/361/1055/table1*

In [11]:
catalog_name = 'J/MNRAS/361/1055/table1'
catalog = Vizier.get_catalogs(catalog_name)
catalog.pprint

<bound method TableList.pprint of TableList with 1 tables:
	'0:J/MNRAS/361/1055/table1' with 12 column(s) and 2446 row(s) >

In [12]:
print(catalog[0].colnames)

['Type', 'OGLE', 'Dist', 'Vmag', 'B-V', 'V-I', 'rms', 'Per', 'Per2', 'Notes', '_RA', '_DE']


- **Type** Types of the star (1,1/2,2,3 or 4)
- **OGLE** Ogle designation ID (HHMMSSss-DDMMSs) J2000
- **Dist** Distance to the nearest star detected with DOPHOT
- **Vmag** V magnitude 
- **B-V** B-V colour index
- **V-I** V-I colour index
- **rms** Scatter of I-band magnitude
- **Per** Period (only for type 3)
- **Per2** Second period for double variable (only type 3)
- **e_Per** rms uncertainty of Per
- **e_Per2** rms uncertainty of Per2

In [13]:
result=Vizier(columns=['Type','OGLE','Vmag','B-V','V-I','rms','Per','Per2']).query_constraints(catalog=catalog_name,OGLE='==05201427-7007382')
result.pprint

<bound method TableList.pprint of TableList with 1 tables:
	'0:J/MNRAS/361/1055/table1' with 8 column(s) and 1 row(s) >

In [14]:
table=result[0]
table

<Table masked=True length=1>
Type       OGLE         Vmag    B-V     V-I     rms     Per     Per2 
                        mag     mag     mag     mag      d       d   
str3      str16       float32 float32 float32 float32 float64 float32
---- ---------------- ------- ------- ------- ------- ------- -------
   4 05201427-7007382  18.278   0.424   0.321   0.129      --      --